src/main/java/edu/internet2/middleware/grouper/changeLog/consumer/Office365ChangeLogConsumer.java

```diff
@@ -26,56 +27,95 @@
-    private final String clientSecret;
-    private final String tenantId;
-    private final String scope;
-    private final String subdomainStem;
-    private final Office365ApiClient apiClient;
+    private  String clientId;
+    private  String clientSecret;
+    private  String tenantId;
+    private  String scope;
+    private String nameOfConsumer;
+    private  Office365ApiClient apiClient;
     private static ScheduledExecutorService scheduledExecutorService;
     public static Map<String, Long> lastScheduledMap;
     private static final long scheduleBuffer = 1000 * 60 * 15;// 15 minutes
 
+    public enum AzureGroupType {Security,Unified,MailEnabled,MailEnabledSecurity}
 
-    private final GrouperSession grouperSession;
+    private GrouperSession grouperSession = null;
 
     public Office365ChangeLogConsumer() {
-        // TODO: this.getConsumerName() isn't working for some reason. track down
-        String name = this.getConsumerName() != null ? this.getConsumerName() : "o365";
+
+    }
+    public Office365ChangeLogConsumer(String nameOfConsumer){
+        this.nameOfConsumer = nameOfConsumer;
+        initConsumer(nameOfConsumer);
+    }
+
+    protected void initConsumer(String name) {
+
         this.clientId = GrouperLoaderConfig.retrieveConfig().propertyValueStringRequired(CONFIG_PREFIX + name + ".clientId");
         this.clientSecret = GrouperLoaderConfig.retrieveConfig().propertyValueStringRequired(CONFIG_PREFIX + name + ".clientSecret");
         this.tenantId = GrouperLoaderConfig.retrieveConfig().propertyValueStringRequired(CONFIG_PREFIX + name + ".tenantId");
         this.scope = GrouperLoaderConfig.retrieveConfig().propertyValueString(CONFIG_PREFIX + name + ".scope", "https://graph.microsoft.com/.default");
-        this.subdomainStem = GrouperLoaderConfig.retrieveConfig().propertyValueString(CONFIG_PREFIX + name + ".subdomainStem", "ksu:NotInLdapApplications:office365:subdomains");
 
         this.grouperSession = GrouperSession.startRootSession();
-        this.apiClient = new Office365ApiClient(clientId, clientSecret, tenantId, scope,  grouperSession);
+
+
+        AzureGroupType groupType = getAzureGroupType(name);
+        edu.internet2.middleware.grouper.changeLog.consumer.model.Group.Visibility visibility = getAzureVisibility(name, groupType);
+
+        this.apiClient = new Office365ApiClient(clientId, clientSecret, tenantId, scope, groupType,visibility, grouperSession);
         if (scheduledExecutorService == null) {
             scheduledExecutorService = Executors.newScheduledThreadPool(1);
         }
         if (lastScheduledMap == null) {
             lastScheduledMap = new ConcurrentHashMap<>();
         }
+    }
 
+    protected edu.internet2.middleware.grouper.changeLog.consumer.model.Group.Visibility getAzureVisibility(String name, AzureGroupType groupType) {
+        edu.internet2.middleware.grouper.changeLog.consumer.model.Group.Visibility visibility = null;
```

Comment:
If edu.internet2.middleware.grouper.changeLog.consumer.model.Group.Visibility doesn't conflict with another class of that name, perhaps it would be better to import it?